maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java

```diff
@@ -335,12 +340,34 @@ private void installOWAs(Server server, DistroProperties distroProperties) throw
     
     private void downloadOWAs(File targetDirectory, DistroProperties distroProperties, File owasDir) throws MojoExecutionException {
         List<Artifact> owas = distroProperties.getOwaArtifacts(distroHelper, targetDirectory);
-        OpenmrsBintray openmrsBintray = new OpenmrsBintray(getProxyFromSettings());
-
         if (!owas.isEmpty()) {
             wizard.showMessage("Downloading OWAs...\n");
             for (Artifact owa: owas) {
-                openmrsBintray.downloadOWA(owasDir, owa.getArtifactId(), owa.getVersion());
+                wizard.showMessage("Downloading OWA: " + owa);
+                installOwa(owasDir, owa);
+            }
+        }
+    }
+
+    private void installOwa(File owasDir, Artifact owa) throws MojoExecutionException {
+        // Support existing Bintray artifacts
+        try {
+            OpenmrsBintray openmrsBintray = new OpenmrsBintray(getProxyFromSettings());
+            openmrsBintray.downloadOWA(owasDir, owa.getArtifactId(), owa.getVersion());
+        }
+        catch (Exception e) {
+            // If file is not found in bintray, and exception will be thrown.  Try to find it in Maven repo.
+            moduleInstaller.installModule(owa, owasDir.getAbsolutePath());
+            File downloadedFile = new File(owasDir, owa.getArtifactId() + "-" + owa.getVersion() + "." + owa.getType());
+            if (!downloadedFile.exists()) {
+                throw new MojoExecutionException("Unable to download OWA from Bintray or Maven", e);
+            }
+            File renamedFile = new File(owasDir, owa.getArtifactId() + OpenmrsBintray.OWA_PACKAGE_EXTENSION);
+            try {
+                FileUtils.moveFile(downloadedFile, renamedFile);
+            }
+            catch (IOException ioe) {
+                throw new MojoExecutionException("Unable to move OWA file to " + renamedFile, e);
```

Comment:
Did you intend to pass `ioe` instead of `e` in `new MojoExecutionException`?

Reply:
Yes, good catch

Reply:
Fixed in latest commit.